src/sherpa_ai/memory/shared_memory_with_vectordb.py

```diff
@@ -0,0 +1,51 @@
+from sherpa_ai.connectors.ChromaVectorStore import ChromaVectorStore
+from sherpa_ai.memory import SharedMemory
+from typing import List, Optional
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+
+from sherpa_ai.actions.planning import Plan
+from sherpa_ai.agents import AgentPool
+from sherpa_ai.events import Event, EventType
+from sherpa_ai.memory.belief import Belief
+import sherpa_ai.config as cfg
+
+
+class SharedMemoryWithVectorDB(SharedMemory):
+    def __init__(
+        self,
+        objective: str,
+        session_id: str,
+        agent_pool: AgentPool = None,
+    ):
+        self.objective = objective
+        self.agent_pool = agent_pool
+        self.events: List[Event] = []
+        self.plan: Optional[Plan] = None
+        self.current_step = None
+        self.session_id = session_id
+
+    def observe(self, belief: Belief):
+        vec_db = ChromaVectorStore.chroma_from_existing()
+
+        tasks = super().get_by_type(EventType.task)
+        task = tasks[-1] if len(tasks) > 0 else None
+        contexts = vec_db.similarity_search(task.content, session_id=self.session_id)
+
+        for context in contexts:
+            print('*******************************************')
+            print(context)
+            print('*******************************************')
+            super().add(
+                agent="planner",
+                event_type=EventType.user_input,
+                content=context.page_content,
+            )
+        belief.set_current_task(task)
+        for event in self.events:
+            if (
```

Comment:
if event.event_type in [EventType.task, EventType.result, EventType.user_input]: